jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/PathMappingsHandler.java

```diff
@@ -0,0 +1,70 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.server.handler;
+
+import org.eclipse.jetty.http.pathmap.MatchedResource;
+import org.eclipse.jetty.http.pathmap.PathMappings;
+import org.eclipse.jetty.http.pathmap.PathSpec;
+import org.eclipse.jetty.server.Handler;
+import org.eclipse.jetty.server.Request;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A Handler that delegates to other handlers through a configured {@link PathMappings}.
+ */
+public class PathMappingsHandler extends Handler.Wrapper
+{
+    private static final Logger LOG = LoggerFactory.getLogger(PathMappingsHandler.class);
+
+    private final PathMappings<Handler> mappings = new PathMappings<>();
+
+    public void addMapping(PathSpec pathSpec, Handler handler)
+    {
+        if (isStarted())
+            throw new IllegalStateException("Cannot add mapping: " + this);
+
+        mappings.put(pathSpec, handler);
```

Comment:
The mapped handlers should still be added to this handler as a bean, so we can dump them even in not started.

Reply:
What about making PathMappings itself be lifecycle-able (for the start/stop of the individual Handlers)?
After all PathMappings is already Dumpable.

Reply:
What about making PathMappings the dump point, and only having doStart/doStop handle the start/stop of the handlers that the PathMappings is tracking? (not as individual beans)